store/sqlstore/user_store.go

```diff
@@ -1471,7 +1471,7 @@ func generateSearchQuery(query sq.SelectBuilder, terms []string, fields []string
 		termArgs := []interface{}{}
 		for _, field := range fields {
 			if isPostgreSQL {
-				searchFields = append(searchFields, fmt.Sprintf("lower(%s) LIKE lower(?) escape '*' ", field))
+				searchFields = append(searchFields, fmt.Sprintf("lower(unaccent(%s)) LIKE lower(unaccent(?)) escape '*' ", field))
```

Comment:
We don't enforce installing extensions on users. They are from the contrib package and it's not mandatory for it to be there. What you can do is change the `LIKE` clause to a `tsvector` search query. Then users can change the default search configuration to use a different dictionary and postgres will pick that up.

Reply:
Ok. One question. How the user is going to change the default search configuration to use different dictionary? Do you mean that the user is going to change them during searching?

Reply:
They can changes the search configuration like this: https://www.postgresql.org/docs/current/unaccent.html#id-1.11.7.54.7. You just need to pass the search configuration name that's stored in `s.pgDefaultTextSearchConfig`.

Reply:
I have made the changes.
I also implemented another method which seemed to make the queries 20 - 50ms longer but is more flexible in the terms
that it always returns all the accented variants using the `translate `function.
```
translationString := "'âãäåāăąÁÂÃÄÅĀĂĄèééêëēĕėęěĒĔĖĘĚìíîïìĩīĭÌÍÎÏÌĨĪĬóôõöōŏőÒÓÔÕÖŌŎŐùúûüũūŭůÙÚÛÜŨŪŬŮ','aaaaaaaaaaaaaaaeeeeeeeeeeeeeeeiiiiiiiiiiiiiiiiooooooooooooooouuuuuuuuuuuuuuuu'"
```
```
searchFields = append(searchFields, fmt.Sprintf("translate(lower(%[1]s),%[2]s) LIKE lower(?) escape '*' OR lower(%[1]s) LIKE lower(?) escape '*'", field, translationString))
```

Reply:
> But the example in the https://github.com/mattermost/mattermost-server/issues/20327 Jérémie is not retrieved when s.pgDefaultTextSearchConfig is pg_catalog.english but when it is e.g. pg_catalog.french . I suppose it is the expected behavior.

Yes, that's the correct behavior.

> It returns more results as it's language agnostic, without using any extension or the default text search configuration.

We are trying to make it language sensitive, not language agnostic.